<launch>

    <arg name="markerId"        default="0"/>
    <arg name="markerSize"      default="0.044"/>    <!-- in m -->
    <arg name="eye"             default="left"/>
    <arg name="marker_frame"    default="marker"/>
    <arg name="ref_frame"       default=""/>  <!-- leave empty and the pose will be published wrt param parent_name -->
    <arg name="corner_refinement" default="LINES" /> <!-- NONE, HARRIS, LINES, SUBPIX -->
    
    
    <!-- right rs d435 : 017322073709 -->
    <!-- <include file="$(find realsense2_camera)/launch/rs_camera.launch">
        <arg name="camera" value="right"/>
        <arg name="serial_no" value="017322073709"/>
    </include> -->

    <!-- front rs d435 : 913422071281 -->
    <!-- <include file="$(find realsense2_camera)/launch/rs_camera.launch">
        <arg name="camera" value="front"/>
        <arg name="serial_no" value="913422071281"/>
    </include> -->
    <!-- left rs d435 : 938422071303 -->
    <include file="$(find realsense2_camera)/launch/rs_camera.launch">
        <arg name="camera" value="left"/>
        <arg name="serial_no" value="938422071303"/>
    </include>

    <node pkg="aruco_ros" type="single" name="aruco_single">
        <!-- <remap from="/camera_info" to="/camera/color/camera_info" />
        <remap from="/image" to="/camera/color/image_raw" /> -->
        <param name="image_is_rectified" value="true"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="marker_id"          value="$(arg markerId)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>   <!-- frame in which the marker pose will be refered -->
        <!-- <param name="camera_frame"       value="stereo_gazebo_$(arg eye)_camera_optical_frame"/> -->
        <param name="use_camera_info" value="true"/>
        <!-- thkim test -->
        <!-- <param name="camera_frame"       value="left_camera_link"/>
        <remap from="/camera_info" to="/mecanum_rb5/left_camera/camera_info" />
        <remap from="/image" to="/mecanum_rb5/left_camera/left_image_raw" /> -->
        
        <param name="marker_frame"       value="$(arg marker_frame)" />
        <param name="corner_refinement"  value="$(arg corner_refinement)" />

        <remap from="/camera_info" to="/camera/color/camera_info" />
        <param name="camera_frame"       value="left_camera_link"/>
        <remap from="/image" to="/camera/color/image_raw" />
    </node>

    <!--- Run Rviz-->
    <node pkg="rviz" type="rviz" name="aruco_rviz" args="-d $(find aruco_ros)/rviz/display_aruco.rviz" />
</launch>

